### PR TITLE
Update the logic in ValidateSagaOutboxCombo and add tests

### PR DESCRIPTION
**Pull request by jpalac**
Fixes https://github.com/Particular/NServiceBus.Persistence.Sql/issues/1492

### CODE REVIEW

**Review by abparticular — APPROVED** [review]

File: src/SqlPersistence/SqlValidateStorageTypeCombinationFeature.cs [hunk 1]
```diff
@@ -29,7 +29,8 @@ static void ValidateSagaOutboxCombo(IReadOnlySettings settings)
         }
         var isSagasEnabledForSqlPersistence = settings.IsFeatureActive(typeof(SqlSagaFeature));
         var isOutboxEnabledForSqlPersistence = settings.IsFeatureActive(typeof(SqlOutboxFeature));
-        if (isSagasEnabledForSqlPersistence && isOutboxEnabledForSqlPersistence)
+        if ((isSagasEnabledForSqlPersistence && isOutboxEnabledForSqlPersistence)
+            || (!isSagasEnabledForSqlPersistence && !isOutboxEnabledForSqlPersistence))
```
Comment (johnsimons) [inline src/SqlPersistence/SqlValidateStorageTypeCombinationFeature.cs:33]: simpler if statement
```suggestion
        if (isSagasEnabledForSqlPersistence == isOutboxEnabledForSqlPersistence)
```

**Review by johnsimons — APPROVED** [review]
Added a small simplification to the code, but overall 👍

File: src/SqlPersistence/SqlValidateStorageTypeCombinationFeature.cs (hunk 1 above)
Comment (adamralph) [reply src/SqlPersistence/SqlValidateStorageTypeCombinationFeature.cs:33]: I was going to send a PR to simplify this but then I noticed this suggestion, which is exactly what I was intending to propose. Any reason this suggestion wasn't accepted?

Comment (jpalac) [reply src/SqlPersistence/SqlValidateStorageTypeCombinationFeature.cs:33]: No reason that I recall apart from "slipped through the cracks"

Comment (adamralph) [reply src/SqlPersistence/SqlValidateStorageTypeCombinationFeature.cs:33]: FWIW I sent https://github.com/Particular/NServiceBus.Persistence.Sql/pull/1525 to improve the exception message, and while I was there I boiled down a lot of the class with refactoring.

**Pull request merged**